src/platform_impl/web/event_loop/runner.rs

```diff
@@ -289,82 +336,178 @@ impl<T: 'static> Shared<T> {
         }
     }
 
-    pub fn handle_scale_changed(&self, old_scale: f64, new_scale: f64) {
-        // If there aren't any windows, then there is nothing to do here.
-        if self.0.all_canvases.borrow().is_empty() {
+    /// Handle a change in scale factor, without any other information.
+    ///
+    /// `initial` is whether this is coming from the initial `ScaleChangeDetector`, rather than the `ResizeObserver`.
```

Comment:
Flags like this is IMHO usually bad practice, perhaps things can be refactored slightly to avoid it? (concretely: I have trouble immediately parsing the `!(initial && observer_will_run)` conditional).